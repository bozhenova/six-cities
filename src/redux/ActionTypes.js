export const ActionTypes = {
  SET_CITY: 'SET_CITY',
  SET_SORT_TYPE: 'SET_SORT_TYPE',
  SET_CURRENT_OFFER: 'SET_CURRENT_OFFER',
  SET_USER_DATA: 'SET_USER_DATA',
  SET_AUTHORIZATION_REQUIRED: 'SET_AUTHORIZATION_REQUIRED',
  SET_ERROR: 'SET_ERROR',
  SET_SENDING_STATUS: 'SET_SENDING_STATUS',

  LOAD_OFFERS: 'LOAD_OFFERS',
  LOAD_NEARBY_OFFERS: 'LOAD_NEARBY_OFFERS',
  LOAD_REVIEWS: 'LOAD_REVIEWS',
  LOAD_FAVORITES: 'LOAD_FAVORITES',

  UPDATE_FAVORITES: 'UPDATE_FAVORITES',

  POST_REVIEW: 'POST_REVIEW',

  LOCK_FORM: 'LOCK_FORM'
};
