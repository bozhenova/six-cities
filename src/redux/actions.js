export const ActionType = {
  CHANGE_CITY: 'CHANGE_CITY'
};

export const ActionCreators = {
  changeCity: city => ({
    type: ActionType.CHANGE_CITY,
    payload: city
  })
};
