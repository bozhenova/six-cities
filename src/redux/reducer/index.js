import { combineReducers } from 'redux';

import { reducer as data } from './data/data';
import { reducer as user } from './user/user';
import { reducer as reviews } from './reviews/reviews';
import { reducer as favorites } from './favorites/favorites';

const rootReducer = combineReducers({
  data,
  user,
  reviews,
  favorites
});

export default rootReducer;
