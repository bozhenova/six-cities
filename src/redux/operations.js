import { ActionCreators } from './actions';
import { Constants } from '../constants';

const Operations = {
  loadOffers: () => (dispatch, _getState, api) => {
    return api
      .get(Constants.HOTEL_PATH)
      .then(response => dispatch(ActionCreators.loadOffers(response)));
  }
};

export default Operations;
