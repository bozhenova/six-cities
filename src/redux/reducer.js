import { ActionType as types } from './actions';
import offers from '../mocks/offers';

const initialState = {
  city: 'Amsterdam',
  offers: [...offers],
  currentOfferId: null,
  sortType: 'popular',
  reviews: {},
  favorites: []
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case types.SET_CITY:
      return { ...state, city: action.payload };
    case types.SET_CURRENT_OFFER:
      return { ...state, currentOfferId: action.payload };
    case types.SET_SORT_TYPE:
      return { ...state, sortType: action.payload };
  }
  return state;
};

export default reducer;
