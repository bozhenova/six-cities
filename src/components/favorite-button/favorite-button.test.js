import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import FavoriteButton from './favorite-button';
import renderer from 'react-test-renderer';
import rootReducer from '../../redux/reducer';

describe('Favorite Button component renders correctly', () => {
  it('should render a favorite button', () => {
    const props = {
      id: 42,
      isFavorite: false,
      prefixClass: 'place-card',
      width: 18,
      height: 19,
      match: {}
    };

    const store = createStore(rootReducer);
    const tree = renderer
      .create(
        <Provider store={store}>
          <FavoriteButton {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
