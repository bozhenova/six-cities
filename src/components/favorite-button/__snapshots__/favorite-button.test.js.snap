// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Button component renders correctly should render a favorite button 1`] = `
<button
  aria-pressed="false"
  className="place-card__bookmark-button  button"
  onClick={[Function]}
  type="button"
>
  <svg
    aria-hidden="true"
    className="place-card__bookmark-icon"
    height={19}
    width={18}
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To
     bookmarks
  </span>
</button>
`;
